File: testdata/fixtures/rb-small/service.rb
# Service layer exercising require_relative + inferred instance-method calls.

require_relative 'models'

MAX_DOGS = 10

def make_dog(name)
  Dog.new(name, "lab")
end

def describe(name)
  d = Dog.new(name, "lab")
  sound = d.speak
  d.breed
  sound
end

File: testdata/fixtures/php-small/service.php
<?php
// Service layer exercising use-imports + new + inferred instance-method calls.

namespace App\Service;

use App\Animals\Dog;

const MAX_DOGS = 10;

function make_dog(string $name): Dog {
    return new Dog($name, "lab");
}

function describe(string $name): string {
    $d = new Dog($name, "lab");
    $sound = $d->speak();
    $d->walk();
    return $sound;
}
